src/sentry/api/endpoints/group_notes_details.py

```diff
@@ -29,11 +32,19 @@ def delete(self, request: Request, group, note_id) -> Response:
         if not request.user.is_authenticated:
             raise PermissionDenied(detail="Key doesn't have permission to delete Note")
 
-        try:
-            note = Activity.objects.get(
-                group=group, type=ActivityType.NOTE.value, user_id=request.user.id, id=note_id
-            )
-        except Activity.DoesNotExist:
+        notes_by_user = Activity.objects.filter(
+            group=group, type=ActivityType.NOTE.value, user_id=request.user.id
+        )
+        if not len(notes_by_user):
+            raise ResourceDoesNotExist
+
+        note = None
+        for note in notes_by_user:
+            if note.id == note_id:
+                note = note
```

Comment:
@ceorourke what are you doing here lol

Reply:
I need to know if the user had left other comments on the issue and then also grab the one comment that matches the `note_id`